content-src/components/ActivityFeed/ActivityFeed.js

```diff
@@ -50,5 +51,48 @@ ActivityFeed.propTypes = {
   length: React.PropTypes.number
 };
 
+function groupSitesByDate(sites) {
+  let groupedSites = {};
+  for (let site of sites) {
+    if (!Number.isInteger(site.lastVisitDate)) {
+      continue;
+    }
+
+    let day = moment(site.lastVisitDate / 1000).startOf("day").format();
+    if (!(day in groupedSites)) {
+      groupedSites[day] = [];
+    }
+    groupedSites[day].push(site);
+  }
+  return groupedSites;
+}
+
+const GroupedActivityFeed = React.createClass({
+  getDefaultProps() {
+    return {length: DEFAULT_LENGTH};
+  },
+  render() {
+    const sites = this.props.sites.slice(0, this.props.length);
+    const groupedSites = groupSitesByDate(sites);
+    const dates = Object.keys(groupedSites).sort(function(a, b) {
+      return moment(b).toDate() - moment(a).toDate();
+    });
+    return (<div className="grouped-activity-feed">
+      {dates.map(date => (<div>
```

Comment:
Sorry, I know I'm annoying and this is a WIP, but I think we'll need a unique `key` prop on the `<div>` otherwise users will see the following warning/error in their console logs:

> "Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `GroupedActivityFeed`. See https://fb.me/react-warning-keys for more information."

![new_tab_and_feat_content___160_date_handling_wip_by_rlr_ _pull_request__197_ _mozilla_activity-streams](https://cloud.githubusercontent.com/assets/557895/13297107/7abe9e44-dae5-11e5-8992-4b6dbd9a4038.png)